### DISCUSSION TITLE
containerlab with openbmp

### DISCUSSION THREAD

**Discussion by ACodingfreak** [srl-labs/containerlab#1666] [Q&A]
Hi All,

Sorry this is not an issue 

I am just checking if anyone tried openbmp (https://www.openbmp.org/) with containerlab ?
If Yes, can you please sample topology file for the same ?

**Comment by hellt** [comment]
Hi
it is not the answer you're looking for but I don't see any blockers to have a lab with openbmp. I haven't heard it being mentioned on our discord, though, so maybe you will be the first one to build it.

**Comment by ACodingfreak** [comment]
Hi @hellt 

Well I tried it out by deploying routers via containerlab and setup openbmp using docker-compose.
Now I am able to connect one of the interfaces on the router where bmpserver configuration is enabled with the obmp-collector container. This is successful and able to bringup the bmp session successfully.

https://www.openbmp.org/getting_started.html
https://raw.githubusercontent.com/OpenBMP/obmp-docker/main/docker-compose.yml 

![image](https://github.com/srl-labs/containerlab/assets/1760137/1ed6241e-0ff6-4150-9918-59af10431b72)

As you can see in docker-compose.yml there are multiple configurations which needs to be provided to set of applications namely obmp-collector, kafka and zookeeper. If all these applications are running in a single container or a VM then i can deploy it using containerlab with kind type as Linux.

But if I have to deploy them as individual containers, how can i share the respective configurations in docker-compose via containerlab topology file ? Any pointers or sample scripts would be of great help.
